biojava-structure/src/main/java/org/biojava/nbio/structure/basepairs/BasePairParameters.java

```diff
@@ -0,0 +1,718 @@
+package org.biojava.nbio.structure.basepairs;
+
+import org.biojava.nbio.structure.*;
+import org.biojava.nbio.structure.contact.Pair;
+import org.biojava.nbio.structure.geometry.SuperPosition;
+import org.biojava.nbio.structure.geometry.SuperPositionQCP;
+import org.biojava.nbio.structure.io.PDBFileReader;
+
+import javax.vecmath.Matrix4d;
+import javax.vecmath.Point3d;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.lang.Math.*;
+
+/**
+ * Contributed to BioJava under its LGPL
+ * This module calculates the el Hassan-Calladine Base Pairing and Base-pair Step Parameters
+ * Citation: https://www.ncbi.nlm.nih.gov/pubmed/11601858
+ *
+ * The method that is usually overridden is findPairs(), this implementation is used for a large-scale
+ * analysis of the most proper helical regions in almost 4000 protein-DNA structures, almost
+ * 2000 structures containing only DNA, or almost 1300 structures containing only RNA. (as of 7/2017).
+ * Those who study tertiary structures for RNA folding should try using the TertiaryBasePairParameters,
+ * because this base class is only looking for base pairs between separate strands that exactly line up.
+ * To relax the lining up policy and allow for non-canonical base pairs, use the MismatchedBasePairParameters
+ * class.
+ *
+ * @author Luke Czapla
+ * @since 5.0.0
+ *
+ */
+public class BasePairParameters implements Serializable {
+
+    private static final long serialVersionUID = 6214502385L;
+    private static Logger log = LoggerFactory.getLogger(BasePairParameters.class);
+
+    // See URL http://ndbserver.rutgers.edu/ndbmodule/archives/reports/tsukuba/Table1.html
+    // and the paper cited at the top of this class (also as Table 1).
+    // These are hard-coded to avoid problems with resource paths.
+    public static String[] standardBases = new String[] {
```

Comment:
All of these constants should be `private static final` and accessed via methods.  The field names should be UPPERCASE_WITH_UNDERSCORES.